app/screens/authentication-screen/pin-screen.tsx

```diff
@@ -0,0 +1,303 @@
+        navigation.reset({
+          index: 0,
+          routes: [
+            { name: 'Primary' }
+          ]
+        })
+      }
+    }
+  }
+
+  const handleCompletedPinForSetPin = (newEnteredPIN: string) => {
+    if (previousPIN.length === 0) {
+      setPreviousPIN(newEnteredPIN)
+      setHelperText(translate("PinScreen.verifyPin"))
+      setEnteredPIN("")
+    } else {
+      verifyPINCodeMathes(newEnteredPIN)
+    }
+  }
+
+  const addDigit = (digit: string) => {
+    if (enteredPIN.length < 4) {
+      let newEnteredPIN = enteredPIN + digit
+
+      setEnteredPIN(newEnteredPIN)
+      console.log(pinAttempts)
+      if (newEnteredPIN.length === 4) {
+        switch (screenPurpose) {
+          case "authenticatePIN": {
+            handleCompletedPinForAuthenticatePin(newEnteredPIN)
+            break
+          }
+          case "setPIN": {
+            handleCompletedPinForSetPin(newEnteredPIN)
+          }
+        }
+      }
+    }
+  }
+
+  const verifyPINCodeMathes = (newEnteredPIN: string) => {
+    if (previousPIN === newEnteredPIN) {
+      RNSecureKeyStore.set("PIN", previousPIN, {accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY})
+        .then((res) => {
+          RNSecureKeyStore.set("pinAttempts", '0', {accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY})
+          navigation.goBack()
+        }, (err) => {
+          returnToSetPin()
+          Alert.alert("Unable to store your pin.")
+        })
+    } else {
+      returnToSetPin()
+    }
+  }
+
+  const returnToSetPin = () => {
+    setPreviousPIN("")
+    setHelperText(translate("PinScreen.setPinFailedMatch"))
+    setEnteredPIN("")
+  }
+
+  const authenticatePINCode = () => {
+
+  }
+
+  return (
+    <Screen style={styles.container} backgroundColor={palette.lightBlue}>
+      <StatusBar
+        backgroundColor={palette.lightBlue}
+        barStyle="light-content" />
+      <View style={styles.topSpacer} />
+      <View style={styles.circles}>
+        <View style={styles.circleContainer}>
+          <View style={enteredPIN.length > 0 ? styles.filledCircle : styles.emptyCircle} />
+        </View>
+        <View style={styles.circleContainer}>
+          <View style={enteredPIN.length > 1 ? styles.filledCircle : styles.emptyCircle} />
+        </View>
+        <View style={styles.circleContainer}>
+          <View style={enteredPIN.length > 2 ? styles.filledCircle : styles.emptyCircle} />
+        </View>
+        <View style={styles.circleContainer}>
+          <View style={enteredPIN.length > 3 ? styles.filledCircle : styles.emptyCircle} />
+        </View>
+      </View>
+      <View style={styles.helperTextContainer}>
+        <Text style={styles.helperText}>{helperText}</Text>
+      </View>
+      <View style={styles.pinPad}>
+        <View style={styles.pinPadRow}>
+          <View style={styles.pinPadButtonContainer}>
+            <Button buttonStyle={styles.pinPadButton} titleStyle={styles.pinPadButtonTitle} title="1" onPress={() => addDigit("1")} />
```

Comment:
Can this repetition be de-duplicated? If you wanted to change something on the `<Button>` tag it looks like you'd have to change every line.
Can't you have a function or component that just takes the number as input and returns the correct `<Button>`